### DISCUSSION TITLE
[Question][Hw3] How to deal with types when declaring functions

### DISCUSSION THREAD

**Discussion by ccloe4** [prosyslab-classroom/cs348-information-security#542] [Q&A]
When I declared dbz_sanitizer function, I wrote the type of arguments like this " [|int;int;int|] "  because the type of line and column number from the debug_location is " int ".

After "make", I got an error message like this.
Error: This expression has type int but an expression was expected of type
         Llvm.llvalue

How can I assign proper types to them ? I want to know how to deal with such types in HW3.
Thank you

**Comment by ccloe4** [comment]
Even if I wrote types like " [|Llvm.i64_type llctx; Llvm.i64_type llctx; Llvm.i64_type llctx|] " the result was same...
How can I fix it ?

**Comment by miraliahmadli** [comment]
int is 32 bits, not 64

**Comment by ccloe4** [comment]
thank you!
